### ISSUE TITLE
Docker Hub support for OCI Artifacts & ORAS

### ISSUE THREAD

**Issue by SteveLasker** [deislabs/oras#129]
@justincormack
The following is content prepared for `./implementors.md`. While Docker Hub now supports OCI Image as a `mediaType` (yeah), it doesn't yet support additional `medaiTypes`, required to represent additional artifacts like Helm, Singularity and other development types.

### [Docker Hub](https://hub.docker.com)  

  ORAS does not provide a default registry. Use the docker.io as the domain for docker hub.

  Docker Hub does not yet accept null config files. It does accept empty `config.json` files with `{}` as the content.

  Docker Hub does not yet accept `mediaTypes` beyond `application/vnd.oci.image.config.v1+json` which is reserved for container images that can be run with `docker run` and `containerd run`.
  
  > All artifacts pushed to docker hub will appear as oci images. 

- Authenticating with Docker Hub

  ```sh
  oras login docker.io -u [user] -p [pwd]
  ```

- Create an empty `config.json file`

  ```sh
  echo "{}" > config.json
  ```

- Pushing Artifacts to Docker Hub

> The following is the syntax that would indicate Docker Hub accepts OCI Artifacts as the `mediaType=application/vnd.unknown.config.v1`. This currently fails.

  ```sh
  oras push docker.io/youruser-org/orastest:1 \
    --manifest-config config.json:application/vnd.unknown.config.v1+json \
    ./artifact.txt:application/vnd.unknown.layer.v1+txt
  ```

The following works, however it pushes the text file as an OCI image, indicating it would be supported by `docker run`, `containerd run` and scanned by security scanners. Which of course would fail.

  ```sh
  oras push docker.io/youruser-org/orastest:1 \
    --manifest-config config.json:application/vnd.oci.image.config.v1+json \
    ./artifact.txt:application/vnd.unknown.layer.v1+txt
  ```

- Pulling Artifacts from Docker Hub

  ```sh
  oras pull docker.io/youruser-org/orastest:1
  ```

**Comment by justincormack** [comment]
It would be really helpful to have some conformance tests.

**Comment by SteveLasker** [comment]
agreed. The tests would need to validate the spec, which i’m waiting on reviewers: https://github.com/opencontainers/artifacts/pull/13
to get it approved. 
This did prompt an addition to support null config which i need to add
I have a separate section i need to add for two options for accepting new manifest.config.mediaTypes. 
1) open - not necessarily the best idea
2) imports a publisherManifest.json that defines the new type. It’s also in the spec. I have more in this PR (https://github.com/opencontainers/artifacts/pull/6) that I’m refactoring into the Spec PR

**Comment by yizha1** [comment]
Docker Hub supporting for OCI artifacts are tracked here https://github.com/docker/roadmap/issues/135

**Comment by thedrow** [comment]
This is now shipped so we can close this issue.

**Comment by shizhMSFT** [comment]
Closed by [Announcing Docker Hub OCI Artifacts Support](https://www.docker.com/blog/announcing-docker-hub-oci-artifacts-support/)
